tools/cdk-build-tools/lib/feature-flag.ts

```diff
@@ -0,0 +1,69 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as semver from 'semver';
+
+type Flags = {[key: string]: any} | undefined
+
+/* eslint-disable jest/no-export */
+
+/**
+ * jest helper function to be used when testing feature flags.
+ *
+ * This function is for writing jest tests that test feature flags that are present in CDKv1
+ * and will be removed in CDKv2. This helps execute tests written for CDKv1 in the context of CDKv2.
+ *
+ * The version of CDK is determined by running `scripts/resolve-version.js`, and the logic is as follows:
+ *
+ * When run in CDKv1, the tests are executed as normal.
+ *
+ * When run in CDKv2, tests that are intended to verify the behaviour when the feature flag is set
+ * (via CDK app context) are executed but without setting the app context. This is because CDKv2
+ * defaults to the behaviour for the flag being enabled.
+ *
+ * When run in CDKv2, tests that are intended to verify the behaviour when the feature flag is not
+ * set are skipped.
+ */
+export function withFeatureFlag<T>(
+  name: string,
+  flags: Flags,
+  cdkApp: new (flags: Flags) => T,
```

Comment:
Is this ever not `cdk.App`? Speculative generality?

Reply:
This parameter is unfortunately required for typescript to let me do instantiate a new `App` inside this method.

Also since this is in `cdk-build-tools`, it cannot take a dependency on `@aws-cdk/core`.